src/vm/methodtablebuilder.cpp

```diff
@@ -3824,6 +3826,14 @@ VOID    MethodTableBuilder::InitializeFieldDescs(FieldDesc *pFieldDescList,
         case ELEMENT_TYPE_R4:
             {
                 dwLog2FieldSize = 2;
+
+                // Call SetHasFloatingPointNumberField() because we find a Single field.
+                if (!hasFloatingPointNumberField)
+                {
+                    GetHalfBakedClass()->SetHasFloatingPointNumberField();
+                    hasFloatingPointNumberField = TRUE;
+                }
```

Comment:
Instead of this `if` I would do just
``` c++
hasFloatingPointNumberField = TRUE;
```
at this place, and add this code after processing all fields:
``` c++
if (hasFloatingPointNumberField) 
    GetHalfBakedClass()->SetHasFloatingPointNumberField(); 
```